kpack-image-builder/controllers/buildworkload_controller_test.go

```diff
@@ -1149,6 +1150,58 @@ var _ = Describe("BuildWorkloadReconciler", func() {
 			})
 		})
 	})
+
+	When("failure during generateDropletStatus call", func() {
+		var (
+			build *buildv1alpha2.Build
+		)
+		BeforeEach(func() {
+			fakeImageConfigGetter.ConfigReturns(image.Config{}, errors.New("fake error"))
+			buildWorkload = buildWorkloadObject(buildWorkloadGUID, namespaceGUID, source, env, services, reconcilerName, buildpacks)
+			Expect(adminClient.Create(ctx, buildWorkload)).To(Succeed())
+
+			createdKpackImage := new(buildv1alpha2.Image)
+			Eventually(func() error {
+				return adminClient.Get(ctx, types.NamespacedName{Name: appGUID, Namespace: namespaceGUID}, createdKpackImage)
+			}).Should(Succeed())
+
+			build = &buildv1alpha2.Build{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "build",
+					Namespace: namespaceGUID,
+					Labels: map[string]string{
+						buildv1alpha2.ImageLabel:           appGUID,
+						buildv1alpha2.ImageGenerationLabel: "1",
+						buildv1alpha2.BuildNumberLabel:     "1",
+					},
+				},
+			}
+
+			Expect(adminClient.Create(ctx, build)).To(Succeed())
+		})
+
+		JustBeforeEach(func() {
+			Expect(k8s.Patch(ctx, adminClient, build, func() {
```

Comment:
Patching the build status could take place in the `BeforeEach` instead of `JustBeforeEach` - this makes the setup simpler to read